tests/test-components/build.rs

```diff
@@ -0,0 +1,75 @@
+use std::{collections::HashMap, path::PathBuf, process::Command};
+
+fn main() {
+    println!("cargo:rerun-if-changed=components");
+    let out_dir = PathBuf::from(std::env::var_os("OUT_DIR").unwrap());
+    let packages = std::fs::read_dir("components")
+        .unwrap()
+        .map(|e| e.unwrap())
+        .map(|e| e.file_name().into_string().unwrap())
+        .collect::<Vec<_>>();
+
+    let mut generated_code = String::new();
+    let mut threads = Vec::new();
+    let mut name_to_path = HashMap::new();
+    for package in packages {
+        let o = out_dir.clone();
+        let manifest_path = PathBuf::from("components")
+            .join(&package)
+            .join("Cargo.toml");
+        let manfiest = cargo_toml::Manifest::from_path(&manifest_path).unwrap();
+
+        let handle = std::thread::spawn(move || {
```

Comment:
Nit: does this actually give us concurrency?  If they're all part of the same workspace, won't all but one end up blocked waiting for the package cache file lock?

Reply:
Yea, you're right. I'll remove.